leadbutt.py

```diff
@@ -218,6 +394,47 @@ def get_metric_statistics(**kwargs):
             output_results(results, this_metric, options)
             time.sleep(kwargs.get('interval', 0) / 1000.0)
 
+        # get enhanced monitoring if it is enabled
+        if enhanced_monitoring:
+            options = get_options(config_options, None, cli_options)
+            # convert minutes to seconds
+            period_local = options['Period'] * 60
+            count_local = options['Count']
+            log_group = enhanced_monitoring['LogGroup']
+            # determine formatter
+            if 'Formatter' in enhanced_monitoring:
+                options['Formatter'] = enhanced_monitoring['Formatter']
+            # determine if there is a custom formatter for logs in list form
+            if 'ListFormatter' in enhanced_monitoring:
+                options['ListFormatter'] = enhanced_monitoring['ListFormatter']
+            else:
+                options['ListFormatter'] = options['Formatter']
+            # if you have metrics that are available only every 5 minutes, be sure to request only stats
+            # that are likely/sure to be up to date, ie ones ending on the previous
+            # period increment.
+            # convert date to miliseconds
+            end_time = int((datetime.datetime.now() - datetime.timedelta(seconds=int(time.time()) % period_local)).strftime("%s")) * 1000
```

Comment:
There is another place in this script where this same interval-guessing happens, can both be refactored to use a function?